tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/iterator/ConsumingIteration.java

```diff
@@ -50,13 +50,21 @@ public ConsumingIteration(CloseableIteration<BindingSet, QueryEvaluationExceptio
 
```

Comment:
Do not really like the idea of fetching solutions from argument iteration within the constructor.  There are possible scenarious where, while iterating, the underlying iteration may block (dueto  a lack of resources/threads in the pools) and the exectution may stall. Better to either postpone it until first `hasNext()` or reintroduce the configurable parameter (from my PR) that we could use to control the `max` size or even disable the caching entirely, if we need to do so ...

Reply:
I have added your changes to make this configurable as a second commit.

Would not change it ad-hoc right now to move the consuming part out of the constructor, I cannot oversee the consequences of that. I remember that this consuming was explicitly required (in the construction part) to avoid too many open connections for small intermediate result sets, i.e. kind of a buffer for the standard case.

If you want to further experiment, please share the details of your evaluations